### PR TITLE
Fix taking over tags from a workaround needle

### PR DESCRIPTION
**Pull request by Martchus**
Should fix: https://progress.opensuse.org/issues/11840
I'm currently still figuring out how to test this.

BTW: Also fixed label which was referencing by name (must reference by ID).

### CODE REVIEW

File: assets/javascripts/needleeditor.js [hunk 1]
```diff
@@ -280,6 +280,7 @@ function loadTagsAndName() {
   $("#needleeditor_tags").find('input').each(function() {
     $(this).prop('checked', tags.indexOf($(this).val()) != -1);
   });
+  $("#property_workaround").prop('checked', $.inArray('workaround', needle.properties) !== -1);
```
Comment (coolo) [inline assets/javascripts/needleeditor.js]: comparing numbers with !== sounds really paranoid :)

Comment (Martchus) [inline assets/javascripts/needleeditor.js]: I usually compare with === and !== for less unexpected conversions.

**Comment by Martchus** [comment]
Any tips on how to proceed with testing this? I would have to add an additional needle fixture which is a workaround needle.

I could also implement https://progress.opensuse.org/issues/9906 first which would ease testing this.

**Comment by coolo** [comment]
Not sure how this would ease it. But you can change the needle editor test to add the property, check it, then create another needle based on it without setting it and then check if it's in that needle too.

**Comment by Martchus** [comment]
The test actually already creates another needle where the workaround property is checked. However, the new needle is not shown in the needle editor without restarting the test. Implementing this is what the mentioned progress item is about.

**Comment by coolo** [comment]
you can just check the json file on disk

**Comment by Martchus** [comment]
Not sure what you mean. For the test the new needle must show up in the needle editor of the web UI. How would I achieve this by just checking the JSON file?
Also see my comment under the progress item for the feature request: https://progress.opensuse.org/issues/9906

**Pull request merged**